gcc12.1 Build Issues (Linux)

## 07 Callback Example

```
../../../examples/07-callback/callback.cpp:283:40: error: pointer ‘_ptr’ used after ‘void* realloc(void*, size_t)’ [-Werror=use-after-free]
  283 |                         bx::debugPrintf("%s(%d): REALLOC %p (old %p) of %d byte(s)\n", _file, _line, ptr, _ptr, _size)
```

Fixed my end by commenting the line. Needs better fix.

## 14 Shadow Volumes Example
```
In copy constructor ‘{anonymous}::Group::Group(const {anonymous}::Group&)’,
    inlined from ‘void std::__new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = {anonymous}::Group; _Args = {const {anonymous}::Group&}; _Tp = {anonymous}::Group]’ at /usr/include/c++/12.1.0/bits/new_allocator.h:175:4,
    inlined from ‘static void std::allocator_traits<std::allocator<_CharT> >::construct(allocator_type&, _Up*, _Args&& ...) [with _Up = {anonymous}::Group; _Args = {const {anonymous}::Group&}; _Tp = {anonymous}::Group]’ at /usr/include/c++/12.1.0/bits/alloc_traits.h:516:17,
    inlined from ‘void std::vector<_Tp, _Alloc>::push_back(const value_type&) [with _Tp = {anonymous}::Group; _Alloc = std::allocator<{anonymous}::Group>]’ at /usr/include/c++/12.1.0/bits/stl_vector.h:1281:30,
    inlined from ‘void {anonymous}::Mesh::load(const char*)’ at ../../../examples/14-shadowvolumes/shadowvolumes.cpp:997:22:
../../../examples/14-shadowvolumes/shadowvolumes.cpp:749:8: error: ‘group.{anonymous}::Group::m_edgePlanes’ may be used uninitialized [-Werror=maybe-uninitialized]
  749 | struct Group
```

Fix by adding this to Group::reset():
```
m_edgePlanes = NULL;
m_aabb={};
m_sphere={};
m_obb={};
```

## 13 - Stencil Example
```
In copy constructor ‘{anonymous}::Group::Group(const {anonymous}::Group&)’,
    inlined from ‘void std::__new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = {anonymous}::Group; _Args = {const {anonymous}::Group&}; _Tp = {anonymous}::Group]’ at /usr/include/c++/12.1.0/bits/new_allocator.h:175:4,
    inlined from ‘static void std::allocator_traits<std::allocator<_CharT> >::construct(allocator_type&, _Up*, _Args&& ...) [with _Up = {anonymous}::Group; _Args = {const {anonymous}::Group&}; _Tp = {anonymous}::Group]’ at /usr/include/c++/12.1.0/bits/alloc_traits.h:516:17,
    inlined from ‘void std::vector<_Tp, _Alloc>::push_back(const value_type&) [with _Tp = {anonymous}::Group; _Alloc = std::allocator<{anonymous}::Group>]’ at /usr/include/c++/12.1.0/bits/stl_vector.h:1281:30,
    inlined from ‘void {anonymous}::Mesh::load(const void*, uint32_t, bgfx::VertexLayout, const uint16_t*, uint32_t)’ at ../../../examples/13-stencil/stencil.cpp:639:21,
    inlined from ‘virtual void {anonymous}::ExampleStencil::init(int32_t, const char* const*, uint32_t, uint32_t)’ at ../../../examples/13-stencil/stencil.cpp:839:20:
../../../examples/13-stencil/stencil.cpp:601:8: error: ‘group.{anonymous}::Group::m_obb’ may be used uninitialized [-Werror=maybe-uninitialized]
  601 | struct Group
      |        ^~~~~
../../../examples/13-stencil/stencil.cpp: In member function ‘virtual void {anonymous}::ExampleStencil::init(int32_t, const char* const*, uint32_t, uint32_t)’:
../../../examples/13-stencil/stencil.cpp:627:23: note: ‘group’ declared here
  627 |                 Group group;
      |                       ^~~~
```

Fix by adding this to Group::reset():
```
m_sphere={};
m_aabb={};
m_obb={};
```

--- a/examples/13-stencil/stencil.cpp
+++ b/examples/13-stencil/stencil.cpp
@@ -609,6 +609,9 @@ struct Group
 	{
 		m_vbh.idx = bgfx::kInvalidHandle;
 		m_ibh.idx = bgfx::kInvalidHandle;
+		m_sphere={};
+		m_aabb={};
+		m_obb={};
 		m_prims.clear();
 	}
 

--- a/examples/14-shadowvolumes/shadowvolumes.cpp
+++ b/examples/14-shadowvolumes/shadowvolumes.cpp
@@ -764,6 +764,10 @@ struct Group
 		m_numEdges = 0;
 		m_edges = NULL;
 		m_edgePlanesUnalignedPtr = NULL;
+		m_edgePlanes = NULL;
+		m_aabb={};
+		m_sphere={};
+		m_obb={};
 		m_prims.clear();
 	}
 

